test/runtests.jl

```diff
@@ -1,5 +1,37 @@
 using Test, Agents, Random
 
+mutable struct Agent0 <: AbstractAgent
+  id::Int
+end
+
+mutable struct Agent1 <: AbstractAgent
+  id::Int
+  pos::Tuple{Int,Int}
+end
+
+mutable struct Agent2 <: AbstractAgent
+  id::Int
+  weight::Float64
+end
+
+mutable struct Agent3 <: AbstractAgent
+  id::Int
+  pos::Tuple{Int,Int}
+  weight::Float64
+end
+
+mutable struct Agent4 <: AbstractAgent
```

Comment:
Maybe some of these could be further combined? I wasn't sure

Reply:
Yeah, in total we need only 4 types of agents for the tests, the multiplicity of with and without position, and with and without some extra parameter `weight` or `a` or whatever.

Reply:
Sorry, I was mistaken. You are correct we need 6 agents, because no position, integer position and tuple position are 3.